compiler/src/dmd/iasmgcc.d

```diff
@@ -191,86 +230,88 @@ int parseExtAsmOperands(Parser)(Parser p, GccAsmStatement s)
-                {
-                    // Skip over opening `[`
-                    p.nextToken();
-                    // Store the symbolic name
-                    name = p.token.ident;
-                    p.nextToken();
-                }
-                else
-                {
-                    p.eSink.error(s.loc, "expected identifier after `[`");
-                    goto Lerror;
-                }
-                // Look for closing `]`
-                p.check(TOK.rightBracket);
-                // Look for the string literal and fall through
-                if (p.token.value == TOK.string_)
-                    goto case;
-                else
-                    goto default;
-
-            case TOK.string_:
-                constraint = p.parsePrimaryExp();
-                if (p.token.value != TOK.leftParenthesis)
-                {
-                    arg = p.parseAssignExp();
-                    p.eSink.error(arg.loc, "`%s` must be surrounded by parentheses", arg.toChars());
-                }
-                else
-                {
-                    // Look for the opening `(`
-                    p.check(TOK.leftParenthesis);
-                    // Parse the assign expression
-                    arg = p.parseAssignExp();
-                    // Look for the closing `)`
-                    p.check(TOK.rightParenthesis);
-                }
-
-                if (!s.args)
-                {
-                    s.names = new Identifiers();
-                    s.constraints = new Expressions();
-                    s.args = new Expressions();
-                }
-                s.names.push(name);
-                s.args.push(arg);
-                s.constraints.push(constraint);
-                numargs++;
-
-                if (p.token.value == TOK.comma)
-                    p.nextToken();
-                break;
+            // Skip over opening `[`
+            p.nextToken();
+            if (p.token.value == TOK.identifier)
+            {
+                // Store the symbolic name
+                name = p.token.ident;
+                p.nextToken();
+            }
+            else
+            {
+                p.eSink.error(p.token.loc, "expected identifier after `[`");
+                goto Lerror;
+            }
+            // Look for closing `]`
+            if (!p.requireToken(TOK.rightBracket))
+                goto Lerror;
+        }
 
-            default:
-                p.eSink.error(p.token.loc, "expected constant string constraint for operand, not `%s`",
-                        p.token.toChars());
+        // Look for the constraint string.
+        Expression constraint;
+        if (p.token.value == TOK.string_)
+        {
+            constraint = p.parsePrimaryExp();
+            if (constraint.isErrorExp())
                 goto Lerror;
         }
+        else
+        {
+            p.eSink.error(p.token.loc, "expected constant string constraint for operand, not `%s`");
```

Comment:
missing format argument

Reply:
Good catch.

Reply:
Pragma(printf) not working as I expect (or it's missing)